internal/repository/userRepository.go

```diff
@@ -29,48 +29,48 @@ func NewUserRepository(client *mongo.Client, dbName string, collectionName strin
 		collection: collection,
 	}
 }
-func (repo *repository) Save(ctx context.Context, user models.User) (models.User, error) {
+func (repo *repository) Save(ctx context.Context, user models.User) (*models.User, error) {
 	logger.Infof("Saving user %s", user.GetFirstName())
 	newUser := mapper.UserToUserEntity(user)
 	_, err := repo.collection.InsertOne(ctx, newUser)
 	if err != nil {
 		logger.Errorf("Error saving user %s: %v", user.GetFirstName(), err)
-		return models.User{}, err
+		return nil, err
 	}
 	logger.Infof("User %s saved successfully", user.GetFirstName())
 	createdUser, err := mapper.UserEntityToUser(newUser)
-	return createdUser, nil
+	return &createdUser, nil
 }
 
-func (repo *repository) GetUserById(ctx context.Context, id string) (models.User, error) {
+func (repo *repository) GetUserById(ctx context.Context, id string) (*models.User, error) {
 	logger.Infof("Getting user by id %s", id)
 	var userEntity entities.UserEntity
 
 	uid, err := uuid.Parse(id)
 	if err != nil {
 		logger.Errorf("Error parsing user id %s: %v", id, err)
-		return models.User{}, err
+		return nil, err
 	}
 
 	err = repo.collection.FindOne(ctx, bson.M{"_id": uid}).Decode(&userEntity)
 	if err != nil {
 		logger.Errorf("Error getting user by id %s: %v", id, err)
-		return models.User{}, err
+		return nil, err
 	}
 	logger.Infof("User %s found successfully", id)
```

Comment:
There need to be two if statements. If the error is an ErrNoDocuments we return nil, nil. If the error is not nil and something other then ErrNoDocument then we return nil, err. ErrNoDocuments was a poor design choice in the Mongo driver. We should handle it and never return it. See the comments in the ps-tag-dev link I posted in another comment.

Reply:
Fixed